lib/liquid/variable_lookup.rb

```diff
@@ -54,6 +54,8 @@ def evaluate(context)
         elsif @command_flags & (1 << i) != 0 && object.respond_to?(key)
           object = object.send(key).to_liquid
 
+        elsif context.strict_variables
```

Comment:
Hm how about moving this into the `else`? Like
```ruby
else
  return nil unless context.strict_variables
  raise Liquid::UndefinedVariable, "undefined variable #{key} in #{object}"
end
```
I think that would be a bit clearer?